engine/crates/federated-graph/src/federated_graph/v4/ids.rs

```diff
@@ -1,10 +1,10 @@
-use super::FederatedGraph;
+use super::*;
 
 macro_rules! id_newtypes {
     ($($storage:ident [ $name:ident ] -> $out:ident,)*) => {
         $(
             #[derive(Debug, Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
-            pub struct $name(usize);
+            pub struct $name(pub usize);
```

Comment:
Don't really mind either way, but why have this pub if we have the `From` impls (or alternatively why the from impls if this is pub?

Reply:
Good catch, the answer is laziness. I wanted to get to a compiling state faster and that helped a lot, but I should finish converting the existing ID constructors and patterns to not use the field. I'll do that and make the inner usize private again before merging.

Reply:
Just pushed that change, let's see what the tests say.